### PR TITLE
cdclog: fix lost delete

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve? 
solve #1063 

### What is changed and how it works?
give the right op for each key.

### Check List 

Tests 

 - Integration test
 - Manual test (add detailed scripts or steps below)

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Release Note

 - Fix the issue that log restore lost delete key during restore.

### CODE REVIEW

**Comment by kennytm** [comment]
please gofmt.

File: pkg/kv/kv.go [hunk 1]
```diff
@@ -57,6 +57,8 @@ type Iter interface {
 	Value() []byte
 	// Close close this iter.
 	Close() error
+	// KeyIsDelete
+	KeyIsDelete() bool
```
Comment (glorv) [inline pkg/kv/kv.go:61]: Shall we replace this with something like `OpType()` so if we need to add more types of keys e.g. `Merge`, we needn't add another method.

Comment (3pointer) [reply pkg/kv/kv.go:61]: @kennytm I want to know what's your thoughts.

Comment (kennytm) [reply pkg/kv/kv.go:61]: +1 for glorv.

Comment (3pointer) [reply pkg/kv/kv.go]: Done, PTAL~

File: pkg/restore/ingester.go [hunk 2]
```diff
@@ -387,12 +387,12 @@ func (i *Ingester) writeToTiKV(
 	regionMaxSize := i.regionSplitSize * 4 / 3
 
 	for iter.Seek(regionRange.Start); iter.Valid() && bytes.Compare(iter.Key(), regionRange.End) <= 0; iter.Next() {
-		size += int64(len(iter.Key()) + len(iter.Value()))
```
Comment (YuJuncen) [inline pkg/restore/ingester.go:390]: Hold on, why remove this 🤔

Comment (glorv) [reply pkg/restore/ingester.go:390]: +1

**Comment by glorv** [comment]
reset LGTM

**Comment by glorv** [comment]
undo LGTM

**Pull request merged**